app/models/user.rb

```diff
@@ -7,58 +7,26 @@ class User < ApplicationRecord
 
   # NOTE: @citizen428 This is temporary code during profile migration and will
   # be removed.
-  # rubocop:disable Metrics/BlockLength
   concerning :ProfileMigration do
     included do
-      PROFIE_FIELDS =
-        (Profiles::ExtractData::DIRECT_ATTRIBUTES + Profiles::ExtractData::MAPPED_ATTRIBUTES.values).freeze
-
       # NOTE: There are rare cases were we want to skip this callback, primarily
       # in tests. `skip_callback` modifies global state, which is not thread-safe
       # and can cause hard to track down bugs. We use an instance-level attribute
       # instead. See `spec/factories/profiles.rb` for an example.
       attr_accessor :_skip_creating_profile
 
       # All new users should automatically have a profile
-      after_create_commit :create_profile, unless: :_skip_creating_profile
+      after_create_commit -> { Profile.create(user: self, data: Profiles::ExtractData.call(self)) },
+                          unless: :_skip_creating_profile
 
       # Keep saving changes locally for the time being, but propagate them to profiles.
       after_update_commit do
-        if (previous_changes.keys.map(&:to_sym) & User::PROFIE_FIELDS).present?
+        if previous_changes.keys.any? { |attribute| attribute.in?(Profile.mapped_attributes) }
           profile.update(data: Profiles::ExtractData.call(self))
         end
       end
-
-      # Define wrapped getters for profile attributes. We first try to get the
```

Comment:
After talking to @Zhao-Andy we decided to remove the getters now. They were always meant to be temporary and they proofed useful while verifying that we are syncing profiles, but they have outlived their usefulness for now. Going forward we will read and write from `User`, but sync to `Profile`. Then at one point we need to make a hard switch and stop acessing these values through `User` completely.